plugins/catalog-backend/src/service/router.ts

```diff
@@ -142,7 +142,62 @@ export async function createRouter(
           );
         }
         res.status(200).json(entities[0]);
-      });
+      })
+
+      .get('/entities/by-uid/:uid/attachments/:key', async (req, res) => {
+        const { uid, key } = req.params;
+        const attachment = await entitiesCatalog.attachment(uid, key);
+
+        if (!attachment) {
+          throw new NotFoundError(
+            `No attachment with key '${key}' found for entity with uid ${uid}, with kind '${kind}' in namespace '${namespace}'`,
+          );
+        }
+
+        // TODO: Include etag for caching
```

Comment:
I think we could optimize downloading here, by introducing etag based caching (this is different from the etag stuff around the ingestion). We could either hash the data in the DB and include an additional field, or just generate a random etag each time the data is updated.